wrappers/src/linklist_cs.cpp

```diff
@@ -46,22 +47,24 @@ REALM_EXPORT void linklist_insert(SharedLinkViewRef* linklist_ptr, size_t link_n
   });
 }
 
-REALM_EXPORT Row* linklist_get(SharedLinkViewRef* linklist_ptr, size_t link_ndx, NativeException::Marshallable& ex)
+REALM_EXPORT Object* linklist_get(SharedLinkViewRef* linklist_ptr, SharedRealm* realm, size_t link_ndx, NativeException::Marshallable& ex)
 {
-  return handle_errors(ex, [&]() -> Row* {
-    const size_t count = (**linklist_ptr)->size();
-    if (link_ndx >= count)
-      throw IndexOutOfRangeException("Get from RealmList", link_ndx, count);
-    auto rowExpr = (**linklist_ptr)->get(link_ndx);
-    return new Row(rowExpr);
-  });
+    return handle_errors(ex, [&]() -> Object* {
+        const size_t count = (**linklist_ptr)->size();
+        if (link_ndx >= count)
+            throw IndexOutOfRangeException("Get from RealmList", link_ndx, count);
+        auto rowExpr = (**linklist_ptr)->get(link_ndx);
+        StringData object_name = ObjectStore::object_type_for_table_name((**linklist_ptr)->get_target_table().get_name());
+        auto object_schema = realm->get()->schema().find(object_name);
+        return new Object(*realm, *object_schema, *new Row(rowExpr));
```

Comment:
Shouldn't it be `Row(rowExpr)` instead of `*new Row(rowExpr)` to avoid leaking it?